### ISSUE TITLE
Validation results are confusing

### ISSUE THREAD

**Issue by kalai2033** [f90/FactorGAN#7]
I have trained your FactorGAN model with following command. 

!python Image2Image.py --cuda --batchSize=2 --loadSize 256 --dataset "fg" --num_joint_samples 366 --factorGAN 1 --experiment_name "fgval"

I have 366 images in my Train directory and 30 images in the Validation directory.

After the training is run successfully, random images are generated in the out/checkpoint/expname/gen/ folder. I had 30 images in the validation directory but 100 images are getting generated.

**Comment by f90** [comment]
Hey! What do you mean with random images? Also did you use your own custom dataset? How many test samples are in this dataset? At the end of the training run it will provide predictions for the test set, not the validation set I think. This is why you see more images there than in the validation set.

**Comment by kalai2033** [comment]
I am using custom dataset but I follow the same structure as Cityscapes dataset. And I don't have any images in the test directory. But I ran the experiment thrice, all the time it generates output for 100 images even though i have only 30 images in the validation directory.

**Comment by kalai2033** [comment]
![image](https://user-images.githubusercontent.com/49688647/84198487-ebc26c00-aaa3-11ea-8ae7-4d5f977b1007.png)
